### ISSUE TITLE
Support JUnit-style composed annotations for Quarkus test-related annotations

### ISSUE THREAD

**Issue by knutwannheden** [quarkusio/quarkus#10297]
JUnit has the concept of [composed annotations and meta-annotations](https://junit.org/junit5/docs/current/user-guide/#writing-tests-meta-annotations) which allows defining custom annotations and declaring JUnit annotations (e.g. `@ExtendWith`) on these custom annotations. When a test class is annotated with the custom annotation it will behave as if it were annotated directly with the JUnit annotation.

I think it would be very useful if this would also be supported by `QuarkusTestExtension` for some of the annotations it uses (e.g. `@QuarkusTestResource` and `@TestProfile`).

**Comment by famod** [comment]
I was just thinking that it would be nice if a `TestProfile` class could be annotated with `@QuarkusTestResource` but I guess this ticket contains a more thorough approach...?

/cc @geoand

**Comment by geoand** [comment]
Yeah, we probably need to think of this more. But I doubt I will be doing that for 1.7, because I'll be taking some time off soon.

**Comment by geoand** [comment]
If someone wants to take a stab at this, I'd be willing to review. Otherwise it's pretty low in my queue of things to do :(

**Comment by kucharzyk** [comment]
@geoand Any news about this ticket?

 I would like to achieve something like this:

```java
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
@TestSecurity(user = "alice", attributes = {
        @SecurityAttribute(key = "name", value = "Alice"),
        @SecurityAttribute(key = "avatar_url", value = "https://avatars.githubusercontent.com/u/95208?v=4"),
        @SecurityAttribute(key = "email", value = ""),
        @SecurityAttribute(key = "login", value = "alice"),
        @SecurityAttribute(key = "external_id", value = "95208"),
})
public @interface TestAsAlice {
}
```

I would be really awesome if I could extract all fields into one place and use meta annotation instead.

**Comment by geoand** [comment]
@kucharzyk thanks for bringing this up, I had totally forgot about it!

https://github.com/quarkusio/quarkus/pull/29970 takes care of the security related annotations.

**Comment by geoand** [comment]
A question for all the folks here: What other test annotations should support this feature?

**Comment by kucharzyk** [comment]
What do you think @geoand about supporting something like @Stereotype annotation in Quarkus?

https://docs.oracle.com/javaee/6/api/javax/enterprise/inject/Stereotype.html

it might be a standard way of solving all problems like this one

**Comment by geoand** [comment]
I don't see how that solves any problems

**Comment by kucharzyk** [comment]
I’ve meant supporting all annotations from stereotype annotations.

**Comment by geoand** [comment]
I might be missing the point, but I still don't see how it's relevant

**Comment by kucharzyk** [comment]
Nevermind. I’ve seen it can be done without that. 

You were asking what other test annotations we should support. For sure also JwtSecurity

**Comment by jeffawx** [comment]
> You were asking what oth

hope `@TestProfile` can support meta-annotation, cause I need to define bunch of test properties in custom profile and reuse.

**Comment by jeffawx** [comment]
> You were asking what oth

hope `@TestProfile` can support meta-annotation, cause I need to define bunch of test properties in custom profile and reuse.
